python/cudf/cudf/utils/ioutils.py

```diff
@@ -322,6 +322,9 @@
 output_as_binary : set, optional, default None
     If a column name is present in the set, that column will be output as
     unannotated binary, rather than the default 'UTF-8'.
+store_schema : bool, default False
```

Comment:
Is there a reason we think this should be False by default? It seems like faithful roundtrips with Arrow would be a benefit by default. However, it seems like enabling this feature will cast / convert some data types (e.g. "days" aren't supported, and only decimal128 is supported -- if I read the rest of this PR correctly). Are those conversions potentially lossy / do they change metadata? If so, are those conversions worth documenting?

Reply:
A couple of reasons why we chose to set it False by default.

One `arrow:schema` should only be used when we want to round-trip certain col types (primarily durations for now) with arrow _only_. Otherwise, cudf roundtrips with itself and arrow perfectly fine.

Second, cudf still supports int96 timestamps as Spark has actively been using it. Enabling `store_schema` breaks Sparks existing and future workflows requiring them to set this to False whenever using int96 timestamps.

Third, like you mentioned, we can't roundtrip decimal32 and decimal64 with cuDF itself with arrow::schema by default without (losslessly) converting them to decimal128.

To summarize, things work perfectly fine without arrow::schema for the most part until we need to faithfully round-trip duration with arrow only in which case it is enabled.

Reply:
Thanks! This kind of information would be good to have in the docstrings!